go-controller/pkg/testing/libovsdb/matchers.go

```diff
@@ -70,21 +70,27 @@ func matchAndReplaceNamedUUIDs(actual, expected []TestData) {
 	uuids := map[string]string{}
 	expectedToActual := map[int]int{}
 	actualFieldsReplaced := map[[2]int]bool{}
-	for i, x := range actual {
-		for j, y := range expected {
+	for i, x := range expected {
+		for j, y := range actual {
+			name, _ := getUUID(x)
+			uuid, _ := getUUID(y)
+			if name == "" || uuid == "" {
+				continue
+			}
 			if !testDataEqual(x, y, true) {
 				continue
 			}
-			uuid := getUUID(x)
-			name := getUUID(y)
 			fname := names[uuid]
 			if fname != "" {
-				panic(fmt.Sprintf("Can't infer named UUIDs, found multiple matches: [%s -> %s, %s]", uuid, name, fname))
+				panic(fmt.Sprintf("Can't infer named UUIDs, multiple expected models match (ignoring uuids) an actual model: [%s -> %s, %s]", uuid, name, fname))
+			}
+			fuuid := uuids[name]
+			if fuuid != "" {
+				panic(fmt.Sprintf("Can't infer named UUIDs, multiple actual models match (ignoring uuids) an expected model: [%s -> %s, %s]", name, uuid, fuuid))
 			}
 			names[uuid] = name
 			uuids[name] = uuid
-			expectedToActual[j] = i
-			break
```

Comment:
Not sure I follow as to why the break needs to be remove?

Reply:
good spot, I don't think it does need to be removed.

Reply:
I want to check all actual against expected to verify there are no unexpected matches (more than 1) between each of them, that would produce unexpected results.

Reply:
So I want to detect all possible instances this could happen:
```
panic(fmt.Sprintf("Can't infer named UUIDs, multiple expected models match (ignoring uuids) an actual model: [%s -> panic(fmt.Sprintf("Can't infer named UUIDs, multiple actual models match (ignoring uuids) an expected model: [%s -> %s, %s]
```
This is because we are inferring correspondence of named uuids to actual uuids by comparing stuff ignoring uuids, and it could be very well possible that two things that should not match actually do when uuids are ignored. This a best effort kind of thing that should not happen in well scoped tests but in case it does is better to detect it and fail.

Reply:
OK